plugins/LOMM/LOMM.cpp

```diff
@@ -0,0 +1,445 @@
+/*
+ * LOMM.cpp
+ *
+ * Copyright (c) 2023 Lost Robot <r94231/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "LOMM.h"
+
+#include "embed.h"
+#include "plugin_export.h"
+
+namespace lmms
+{
+
+extern "C"
+{
+
+Plugin::Descriptor PLUGIN_EXPORT lomm_plugin_descriptor =
+{
+	LMMS_STRINGIFY(PLUGIN_NAME),
+	"LOMM",
+	QT_TRANSLATE_NOOP("PluginBrowser", "Upwards/downwards multiband compression plugin powered by the eldritch elder god LOMMUS."),
+	"Lost Robot <r94231/at/gmail/dot/com>",
+	0x0100,
+	Plugin::Type::Effect,
+	new PluginPixmapLoader("logo"),
+	nullptr,
+	nullptr
+};
+
+}
+
+
+LOMMEffect::LOMMEffect(Model* parent, const Descriptor::SubPluginFeatures::Key* key) :
+	Effect(&lomm_plugin_descriptor, parent, key),
+	m_lommControls(this),
+	m_sampleRate(Engine::audioEngine()->processingSampleRate()),
+	m_lp1(m_sampleRate),
+	m_lp2(m_sampleRate),
+	m_hp1(m_sampleRate),
+	m_hp2(m_sampleRate),
+	m_needsUpdate(true),
+	m_coeffPrecalc(-0.05),
+	m_crestTimeConst(0.999),
+	m_lookWrite(0),
+	m_lookBufLength(2)
+{
+	autoQuitModel()->setValue(autoQuitModel()->maxValue());
+	
+	m_yL[0][0] = m_yL[0][1] = LOMM_MIN_FLOOR;
+	m_yL[1][0] = m_yL[1][1] = LOMM_MIN_FLOOR;
+	m_yL[2][0] = m_yL[2][1] = LOMM_MIN_FLOOR;
+	
+	connect(Engine::audioEngine(), SIGNAL(sampleRateChanged()), this, SLOT(changeSampleRate()));
+	emit changeSampleRate();
+}
+
+void LOMMEffect::changeSampleRate()
+{
+	m_sampleRate = Engine::audioEngine()->processingSampleRate();
+	m_lp1.setSampleRate(m_sampleRate);
+	m_lp2.setSampleRate(m_sampleRate);
+	m_hp1.setSampleRate(m_sampleRate);
+	m_hp2.setSampleRate(m_sampleRate);
+	
+	m_coeffPrecalc = -2.2f / (m_sampleRate * 0.001f);
+	m_needsUpdate = true;
+	
+	m_crestTimeConst = exp(-1.f / (0.2f * m_sampleRate));
+	
+	m_lookBufLength = std::ceil((LOMM_MAX_LOOKAHEAD / 1000.f) * m_sampleRate) + 2;
+	for (int i = 0; i < 2; ++i)
+	{
+		for (int j = 0; j < 3; ++j)
```

Comment:
I recommend using more descriptive name for loop variables, e.g., `ch` and `band`.

Reply:
Normally I'd agree, but the *large* majority of variable usage throughout this function uses `i` and `j` as array indices, so changing them to more descriptive labels results in a surprisingly large deficit in the readability of the entire code.  I added comments to the loops to make sure the variables' purposes are obvious, let me know if you still think this change should be made.